DirectProgramming/DPC++/StructuredGrids/1d_HeatTransfer/src/1d_HeatTransfer.cpp

```diff
@@ -177,50 +240,18 @@ float* ComputeHeatHostSerial(float* arr, float* arr_next, float C,
     arr_next[num_p + 1] = arr[num_p];
 
     // Swap the buffers at every iteration.
-    swap = arr;
+    float* swap = arr;
     arr = arr_next;
     arr_next = swap;
   }
 
   return arr;
 }
 
-//************************************
-// Function description: calculate the results computed by the host and by the
-// device.
-//************************************
-bool CompareResults(float* device_results, float* host_results,
-                    unsigned int num_point, float C) {
-  float delta = 0.001f;
-  float difference = 0.00f;
-  double norm2 = 0;
-  bool err = false;
-
-  ofstream err_file;
-  err_file.open("error_diff.txt");
-
-  err_file << " \t idx\theat[i]\t\theat_CPU[i] \n";
-
-  for (unsigned int i = 0; i < num_point + 2; i++) {
-    err_file << "\n RESULT: " << i << "\t" << std::setw(12) << std::left
-             << device_results[i] << "\t" << host_results[i];
-
-    difference = fabsf(host_results[i] - device_results[i]);
-    norm2 += difference * difference;
-
-    if (difference > delta) {
-      err = true;
-      err_file << ", diff: " << difference;
-    }
-  }
-
-  return err;
-}
 
 int main(int argc, char* argv[]) {
-  unsigned int n_point;  // The number of point in 1D space
-  unsigned int
-      n_iteration;  // The number of iteration to simulate the heat propagation
+  unsigned int n_point;      // The number of point in 1D space
```

Comment:
Would be more concise to use size_t instead of unsigned int.   https://en.cppreference.com/w/cpp/types/size_t

Reply:
Changed to size_t